src/cdtools/datasets/ptycho_2d_dataset.py

```diff
@@ -398,3 +398,47 @@ def downsample(self, factor=2):
                 self.background.unsqueeze(0).unsqueeze(0),
                 factor,
                 divisor_override=1)[0,0]
+
+
+    def crop_2D_translations(self, roi):
+        """Shrinks the range of translation positions that are analyzed
+        
+        This deletes all diffraction patterns associated with x- and 
+        y-translations that lie outside of a specified rectangular
+        region of interest. In essence, this operation crops the "relative 
+        displacement map" (shown in self.inspect()) down to the region of 
+        interest.
+
+        Parameters:
+        ----------
+        roi : tuple(float, float, float, float)
+            The translation-x and -y coordinates that define the rectangular 
+            region of interest as (in units of meters)
+            ('left-most x', 'right-most x', 'top-most y', 'bottom-most y').
```

Comment:
I think it'd be more aligned with other scientific-python-world functions to do (left, right, bottom, top), a-la matplotlib's imshow.

Maybe it would be good to just add a line at the start so that if, e.g., top < bottom or left < right, it just swaps them . I think that'd avoid a lot of frustration when it's being used (especially because the convention for displaying the translations is already so messed up, with positive x-values on the left). Seem reasonable?

Reply:
I think these are really good suggestions for making this method easier to use. I've modified the code accordingly to also ensure that 'top < bottom' and 'left < right' is enforced.